src/main/java/domain/Lotto.java

```diff
@@ -1,57 +1,68 @@
 package domain;
 
-import java.util.*;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
-public class Lotto {
+import static util.LottoNumberGenerator.LOTTO_NUMBER_COUNT_LIMIT;
 
-    private static final int LOTTO_NUMBER_COUNT_LIMIT = 6;
+public class Lotto {
     private Set<LottoNumber> lottoNumbers;
 
-    public Lotto(Integer... input) {
-        this(Arrays.stream(input).collect(Collectors.toList()));
+    public Lotto(int num1, int num2, int num3, int num4, int num5, int num6) {
+        this(IntStream.of(num1, num2, num3, num4, num5, num6)
+                .mapToObj(LottoNumber::new)
+                .collect(Collectors.toSet()));
     }
 
     public Lotto(String input) {
         this(Arrays.stream(input.split(","))
                 .map(String::trim)
                 .map(Integer::parseInt)
-                .collect(Collectors.toList()));
+                .map(LottoNumber::new)
+                .collect(Collectors.toSet()));
```

Comment:
map이 3번 이어지고 있는데 정적 팩토리 함수를 사용하면 이를 좀 줄여볼 수 있을 것 같아요 ㅎㅎ 

```suggestion
    .map(LottoNumber::from)
    .collect(Collectors.toSet()));
}

...

public class LottoNumber {
    ...
    
    public static LottoNumber from(String input) {
        int num = Integer.parseInt(input);
        return LottoNumber.valueOf(num); // 캐싱된 로또 번호를 가져온다. 
    }

}
```

Reply:
정적 팩토리 함수와 valueOf 를 계속 말씀해주셔도 감이 안왔었는데..
이 예시로 보니까 확 이해가 되네요 ㅠ_ㅠ ㅎㅎ 감사합니다 ☺️